src/Compilers/Core/Portable/StrongName/DesktopStrongNameProvider.cs

```diff
@@ -386,21 +347,28 @@ private unsafe void Sign(string filePath, ImmutableArray<byte> keyPair)
             }
         }
 
+        public override int GetHashCode()
+        {
+            return Hash.CombineValues(_keyFileSearchPaths, StringComparer.Ordinal);
+        }
+
         public override bool Equals(object obj)
         {
-            // Explicitly check that we're not comparing against a derived type
-            if (obj == null || GetType() != obj.GetType())
+            if (obj is null || GetType() != obj.GetType())
             {
                 return false;
             }
 
             var other = (DesktopStrongNameProvider)obj;
-            return _keyFileSearchPaths.SequenceEqual(other._keyFileSearchPaths, StringComparer.Ordinal);
-        }
-
-        public override int GetHashCode()
-        {
-            return Hash.CombineValues(_keyFileSearchPaths, StringComparer.Ordinal);
+            if (!FileSystem.Equals(other.FileSystem))
+            {
+                return false;
+            } 
```

Comment:
Can use `FileSystem != other.FileSystem` since reference equality should be sufficient.

Reply:
Done.